src/bench/locustfile.py

```diff
@@ -0,0 +1,219 @@
+"""QualiCharge API load tests."""
+
+import os
+from datetime import datetime, timedelta, timezone
+from functools import cache
+from pathlib import Path
+
+import pandas as pd
+import requests
+from locust import FastHttpUser, task
+
+API_ADMIN_USER = "admin"
+API_ADMIN_PASSWORD = "admin"
+STATIC_DB_OFFSET = 500
+
+
+@cache
+def load_statique_db(db_path: Path):
+    """Load statique database."""
+    db = pd.read_json(db_path, dtype_backend="pyarrow", lines=True)
+    return db.to_dict(orient="records")
+
+
+class BaseAPIUser(FastHttpUser):
+    """Base API user."""
+
+    abstract: bool = True
+    username: str
+    password: str
+    counter: dict = {
+        "static": {"create": STATIC_DB_OFFSET, "bulk": 10000},
+        "status": 0,
+        "session": 0,
+    }
+
+    def on_start(self):
+        """Log user in on start."""
+        credentials = {
+            "username": self.username,
+            "password": self.password,
+        }
+        response = requests.post(f"{self.host}/auth/token", data=credentials)
+        token = response.json()["access_token"]
+        self.client.auth_header = f"Bearer {token}"
+        self.statique_db = load_statique_db(
+            Path(os.environ["QUALICHARGE_STATIQUE_DATA_PATH"])
+        )
+
+    @task
+    def whoami(self):
+        """Test the /whoami endpoint."""
+        self.client.get("/auth/whoami")
+
+    @task
+    def statique_list(self):
+        """Assess the /statique GET endpoint."""
+        with self.rest("GET", "/statique/?limit=100") as response:
+            if response.js["size"] == 0:
+                response.failure("Database contains no statique entries")
+        with self.rest("GET", "/statique/?limit=10") as response:
+            if response.js["size"] == 0:
+                response.failure("Database contains no statique entries")
+
+    @task
+    def statique_get(self):
+        """Assess the /statique/{id_pdc_itinerance} GET endpoint."""
+        id_pdc_itinerance = "FRALLEGO002006P3"
+        with self.rest("GET", f"/statique/{id_pdc_itinerance}") as response:
+            if "id_pdc_itinerance" not in response.js:
+                response.failure(
+                    f"Database does not contain target statique entry {id_pdc_itinerance}"
+                )
+
+    @task
+    def statique_update(self):
+        """Assess the /statique/{id_pdc_itinerance} PUT endpoint."""
+        data = self.statique_db[22]
+        data["date_maj"] = "2024-11-21"
```

Comment:
```suggestion
        data["date_maj"] = str(date.today())
```

Reply:
Same here. We don't care as long as `date_maj` is greater than the database value.